### PR TITLE
Add create_response mutation to GraphQL

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
Add create_response mutation to GraphQL
 
## Insight
- Remove test mutation and add this
- Create [custom inputs](https://graphql-ruby.org/type_definitions/input_objects.html) and they are used for response attributes
- Add tests
 

## Proof Of Work
- Tests should be passed
- Create success with valid attributes
Altair

- Show error message when given invalid attributes
Altair

### CODE REVIEW

File: app/graphql/mutations/create_response_mutation.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    description "Create a new survey's response"
+
+    argument :response, Types::Inputs::ResponseInput, required: true
+
+    field :success, Boolean, null: false
+
+    def resolve(response:)
+      response_form = ResponseForm.new(response)
+
+      raise(GraphQL::ExecutionError, response_form.errors.full_messages.to_sentence) unless response_form.save
```
Comment (team-nimblehq) [inline app/graphql/mutations/create_response_mutation.rb:14]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/096a0b3446faa9f2f3f01faae7a30306d5fe7136/app/graphql/mutations/create_response_mutation.rb#L14"></span>Refers to ‘response_form’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: .reek.yml [hunk 2]
```diff
@@ -30,6 +30,9 @@ directories:
       enabled: false
     InstanceVariableAssumption:
       enabled: false
+  "app/graphql":
+    FeatureEnvy:
+      enabled: false
```
Comment (olivierobert) [inline .reek.yml:35]: Why is this needed?

File: app/graphql/mutations/create_response_mutation.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+
+      {
+        success: true
+      }
```
Comment (olivierobert) [inline app/graphql/mutations/create_response_mutation.rb:18]: Feels a bti awkward to just return `success: true`. In the REST API, we just send 201 so I understand where this comes from but still returning the actual data saved seems more standard as per the [official doc](https://graphql.org/learn/queries/#mutations).

File: spec/graphql/mutations/create_response_mutation_spec.rb [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+RSpec.describe Mutations::CreateResponseMutation do
+  context 'given valid attributes' do
+    it 'returns success' do
+      variables = {
+        input: {
+          response: {
+            surveyId: 'd5de6a8f8f5f1cfe51bc',
+            questions: [
+              {
+                id: '940d229e4cd87cd1e202'
```
Comment (olivierobert) [inline spec/graphql/mutations/create_response_mutation_spec.rb:14]: Why this question has no answer?

**Review by olivierobert — APPROVED** [review]
Looks good 💪
